applications/GeoMechanicsApplication/python_scripts/apply_vector_constraint_table_process.py

```diff
@@ -1,67 +1,82 @@
-            self.x_params.AddValue("value",settings["value"][0])
-            self.x_params.AddEmptyValue("variable_name").SetString(variable_name+"_X")
-            if settings["table"][0].GetInt() == 0:
-                self.components_process_list.append(KratosMultiphysics.ApplyConstantScalarValueProcess(self.model_part, self.x_params))
-            else:
-                self.x_params.AddValue("table",settings["table"][0])
-                self.components_process_list.append(KratosGeo.ApplyComponentTableProcess(self.model_part, self.x_params))
-
-        if settings["active"][1].GetBool() == True:
-            self.y_params = KratosMultiphysics.Parameters("{}")
-            self.y_params.AddValue("model_part_name",settings["model_part_name"])
-            if settings.Has("is_fixed"):
-                self.y_params.AddValue("is_fixed",settings["is_fixed"][1])
-            self.y_params.AddValue("value",settings["value"][1])
-            self.y_params.AddEmptyValue("variable_name").SetString(variable_name+"_Y")
-            if settings["table"][1].GetInt() == 0:
-                self.components_process_list.append(KratosMultiphysics.ApplyConstantScalarValueProcess(self.model_part, self.y_params))
-            else:
-                self.y_params.AddValue("table",settings["table"][1])
-                self.components_process_list.append(KratosGeo.ApplyComponentTableProcess(self.model_part, self.y_params))
-
-        if settings["active"][2].GetBool() == True:
-            self.z_params = KratosMultiphysics.Parameters("{}")
-            self.z_params.AddValue("model_part_name",settings["model_part_name"])
-            if settings.Has("is_fixed"):
-                self.z_params.AddValue("is_fixed",settings["is_fixed"][2])
-            self.z_params.AddValue("value",settings["value"][2])
-            self.z_params.AddEmptyValue("variable_name").SetString(variable_name+"_Z")
-            if settings["table"][2].GetInt() == 0:
-                self.components_process_list.append(KratosMultiphysics.ApplyConstantScalarValueProcess(self.model_part, self.z_params))
-            else:
-                self.z_params.AddValue("table",settings["table"][2])
-                self.components_process_list.append(KratosGeo.ApplyComponentTableProcess(self.model_part, self.z_params))
 
-    def ExecuteInitialize(self):
+    def _CreateParametersForComponent(self, settings, component):
+        index = ComponentToIndex(component)
+        if not settings["active"][index].GetBool():
+            return None
+
+        parameters = KratosMultiphysics.Parameters("{}")
+        parameters.AddValue("model_part_name", settings["model_part_name"])
+        if settings.Has("is_fixed"):
+            parameters.AddValue("is_fixed", settings["is_fixed"][index])
+        parameters.AddValue("value", settings["value"][index])
+        variable_name = settings["variable_name"].GetString() + "_" + component
```

Comment:
This:
`settings["variable_name"].GetString() + "_" + component`
Can be improved to this:
`f"{settings["variable_name"].GetString()}_{component}"`

Reply:
I agree that f-strings should be preferred over traditional string concatenations. I have adopted your suggestion in a slightly different form.